### DISCUSSION TITLE
Resource<T> vs Resource<T | undefined>

### DISCUSSION THREAD

**Discussion by yw662** [solidjs/solid#988] [Polls]
Isn't every `Resource` practically a `Resource<T | undefined>`, since the data needs to be fetched before usable ?
So why not `Resource<T>` extending from `Accessor<T | undefined>` instead ? ([`signal.ts`, line 395](https://github.com/solidjs/solid/blob/be6623d10d11fb28743a074c31e785c18e95bd25/packages/solid/src/reactive/signal.ts#L395)) ?

**Comment by Exelord** [comment]
Resource is a generic type. `createResource` is using `<T | undefined>` as a generic for Resource which ends with the expected result :) 

<img width="545" alt="Screenshot 2022-05-19 at 21 19 20" src="https://user-images.githubusercontent.com/1469375/169385707-dd37c4ea-222a-4f3e-80ed-13bbff5dbccd.png">


The Resource type itself has no assumption about async state which allows to use it for multiple purposes respecting the interface.

**Comment by cstrnt** [comment]
`createResource` also takes in the initialValue and therefore `Resource` can in fact be `T`
